sdk/wallet-adapter/adapters/base-adapter/src/index.ts

```diff
@@ -27,16 +27,22 @@ export interface WalletAdapter {
     event: E,
     callback: WalletAdapterEvents[E]
   ) => () => void;
-  signTransaction(transaction: SignableTransaction): Promise<SignedTransaction>;
+  signTransaction(transactionInput: {
+    transaction: SignableTransaction;
+    account: string;
+    chain: string;
+  }): Promise<SignedTransaction>;
   /**
    * Suggest a transaction for the user to sign. Supports all valid transaction types.
    */
-  signAndExecuteTransaction(
-    transaction: SignableTransaction,
+  signAndExecuteTransaction(transactionInput: {
```

Comment:
I think these could just be wallet standard input types instead of having them mirrored from there.